### ISSUE TITLE
Upgrading from 3.5.3 to 3.5.7 fails

### ISSUE THREAD

**Issue by aaronjwood** [neo4j/docker-neo4j#212]
- Neo4j version: 3.5.3 -> 3.5.7
- Operating system: [Debian](https://github.com/kubernetes/kops/blob/master/docs/images.md#debian)
- API/Driver: Cypher
- Environment: Official neo4j container running on kubernetes 1.11.8

## Expected behavior:
Use a stateful set when deploying neo4j, deploy neo4j at version 3.5.3, upgrade the stateful set to 3.5.7 and everything should work.

## Actual behavior: 
After upgrading the container continually crashes with the following logs:
```
Active database: graph.db
Directories in use:
  home:         /var/lib/neo4j
  config:       /var/lib/neo4j/conf
  logs:         /logs
  plugins:      /var/lib/neo4j/plugins
  import:       /var/lib/neo4j/import
  data:         /var/lib/neo4j/data
  certificates: /var/lib/neo4j/certificates
  run:          /var/lib/neo4j/run
Starting Neo4j.
2019-07-08 19:33:11.901+0000 WARN  Unknown config option: causal_clustering.discovery_listen_address
2019-07-08 19:33:11.904+0000 WARN  Unknown config option: causal_clustering.raft_advertised_address
2019-07-08 19:33:11.904+0000 WARN  Unknown config option: causal_clustering.raft_listen_address
2019-07-08 19:33:11.904+0000 WARN  Unknown config option: ha.host.coordination
2019-07-08 19:33:11.905+0000 WARN  Unknown config option: causal_clustering.transaction_advertised_address
2019-07-08 19:33:11.905+0000 WARN  Unknown config option: causal_clustering.discovery_advertised_address
2019-07-08 19:33:11.906+0000 WARN  Unknown config option: ha.host.data
2019-07-08 19:33:11.906+0000 WARN  Unknown config option: causal_clustering.transaction_listen_address
2019-07-08 19:33:11.929+0000 INFO  ======== Neo4j 3.5.7 ========
2019-07-08 19:33:11.942+0000 INFO  Starting...
2019-07-08 19:33:13.002+0000 ERROR Failed to start Neo4j: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@38604b81' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to obtain lock on store lock file: /data/databases/store_lock. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)". Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@38604b81' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to obtain lock on store lock file: /data/databases/store_lock. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)".
org.neo4j.server.ServerStartupException: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@38604b81' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to obtain lock on store lock file: /data/databases/store_lock. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)".
	at org.neo4j.server.exception.ServerStartupErrors.translateToServerStartupError(ServerStartupErrors.java:45)
	at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:187)
	at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:124)
	at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:91)
	at org.neo4j.server.CommunityEntryPoint.main(CommunityEntryPoint.java:32)
Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.server.database.LifecycleManagingDatabase@38604b81' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to obtain lock on store lock file: /data/databases/store_lock. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)".
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:473)
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:111)
	at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:180)
	... 3 more
Caused by: java.lang.RuntimeException: Error starting org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory, /data/databases
	at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:230)
	at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.newFacade(GraphDatabaseFacadeFactory.java:146)
	at org.neo4j.server.database.CommunityGraphFactory.newGraphDatabase(CommunityGraphFactory.java:41)
	at org.neo4j.server.database.LifecycleManagingDatabase.start(LifecycleManagingDatabase.java:90)
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:452)
	... 5 more
Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter@54eb2b70' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to obtain lock on store lock file: /data/databases/store_lock. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)".
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:473)
	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:111)
	at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:225)
	... 9 more
Caused by: org.neo4j.kernel.StoreLockException: Unable to obtain lock on store lock file: /data/databases/store_lock. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)
	at org.neo4j.kernel.internal.locker.StoreLocker.storeLockException(StoreLocker.java:116)
	at org.neo4j.kernel.internal.locker.StoreLocker.unableToObtainLockException(StoreLocker.java:109)
	at org.neo4j.kernel.internal.locker.StoreLocker.checkLock(StoreLocker.java:97)
	at org.neo4j.kernel.internal.locker.GlobalStoreLocker.checkLock(GlobalStoreLocker.java:61)
	at org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter.start(StoreLockerLifecycleAdapter.java:36)
	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:452)
	... 11 more
2019-07-08 19:33:13.003+0000 INFO  Neo4j Server shutdown initiated by request
```

**Comment by jennyowen** [comment]
This seems to be the same as issue #207 which was caused by some necessary changes to file permissions that I did for 3.5.6. There were a lot of complaints about neo4j containers messing with file permission access on mounted folders, which was fixed in 3.5.6.
I wrote a justification for the change at the time here: https://github.com/neo4j/docker-neo4j/issues/131#issuecomment-489612338 

I have questions for anyone who is hitting this issue:

1) What folders are being mounted when this problem happens? 
2) what user is the owner of the *host* folder(s) and what user is docker running as?
3) does running docker with `--user="$(id -u):$(id -g)"`  fix the problem?
4) does running with `--user="0:0"` fix the problem? (this seemed to work for another user: https://github.com/neo4j/docker-neo4j/issues/207#issuecomment-558149437)

**Comment by jennyowen** [comment]
OK nevermind, I think I've figured this out.
I've been able to reproduce this by mounting the `/data` folder on `neo4j:3.5.3` and writing some small amount of data. Then closing the container and mounting the same folder to `/data` on `neo4j:3.5.7`.
The entrypoint script is *supposed* to check if the `/data` folder is writable and then `chown -R neo4j:neo4j` if not (and we're not running as any specific user).

The problem seems to be that the `/data/databases` folder is owned by the correct `neo4j` *group* but not the `neo4j` owner. HOWEVER instead of having permissions `777` like `/data` it has permissions `755` which prevents group write access. Our writability check checks to see if the running user has `group` access  but doesn't actually check that the folder has permissions allowing group write.

I'll create a pull request to fix it.
